config/sources/odroidxu4.conf

```diff
@@ -1,53 +1,97 @@
-
-BOOTSOURCE='https://github.com/hardkernel/u-boot.git'
-BOOTBRANCH='branch:odroidxu3-v2012.07'
 BOOTDIR='u-boot-odroidxu'
+
 BOOTSIZE=64
-OFFSET=2
-UBOOT_NEEDS_GCC='< 4.9'
+OFFSET=4
 
 case $BRANCH in
 	default)
+
+	BOOTSOURCE='https://github.com/hardkernel/u-boot.git'
+	BOOTBRANCH='branch:odroidxu3-v2012.07'
+	
 	KERNELSOURCE='https://github.com/hardkernel/linux'
 	KERNELBRANCH='branch:odroidxu3-3.10.y'
 	KERNELDIR='linux-odroidxu4'
+	UBOOT_NEEDS_GCC='< 4.9'
+
 	;;
 
 	next)
+
+	BOOTSOURCE=$MAINLINE_UBOOT_SOURCE		
+	BOOTBRANCH='tag:v2016.05' # $MAINLINE_UBOOT_BRANCH
+
+	BOOTCONFIG=odroid-xu4_defconfig
+
 	KERNELSOURCE=$MAINLINE_KERNEL_SOURCE
 	KERNELBRANCH=$MAINLINE_KERNEL_BRANCH
         KERNELDIR=$MAINLINE_KERNEL_DIR
 	;;
 esac
 
 CPUMIN=600000
-CPUMAX=2000000
-GOVERNOR=ondemand
+CPUMAX=1600000
 
 SERIALCON=ttySAC2
 
-UBOOT_FILES="sd_fuse/hardkernel/bl1.bin.hardkernel sd_fuse/hardkernel/bl2.bin.hardkernel u-boot.bin sd_fuse/hardkernel/tzsw.bin.hardkernel"
+if [[ $BRANCH == next ]] ; then
+	UBOOT_FILES="xu4_blobs/bl1.bin.hardkernel xu4_blobs/bl2.bin.hardkernel.1mb_uboot u-boot-dtb.bin xu4_blobs/tzsw.bin.hardkernel"
+else 
+	UBOOT_FILES="sd_fuse/hardkernel/bl1.bin.hardkernel sd_fuse/hardkernel/bl2.bin.hardkernel u-boot.bin sd_fuse/hardkernel/tzsw.bin.hardkernel"
+fi	
 
 write_uboot_platform()
 {
-	dd if=$1/bl1.bin.hardkernel of=$2 seek=1 conv=fsync > /dev/null 2>&1
-	dd if=$1/bl2.bin.hardkernel of=$2 seek=31 conv=fsync > /dev/null 2>&1
-	dd if=$1/u-boot.bin of=$2 bs=512 seek=63 conv=fsync > /dev/null 2>&1
-	dd if=$1/tzsw.bin.hardkernel of=$2 seek=719 conv=fsync > /dev/null 2>&1
-	dd if=/dev/zero of=$2 seek=1231 count=32 bs=512 conv=fsync > /dev/null 2>&1
+	if [[ $BRANCH == next ]] ; then
```

Comment:
$BRANCH here won't be available when installing .deb package with u-boot, so this has to be differentiated i.e. based on u-boot file name